### PR TITLE
Splitting fluid-framework api docs based on exported packages.

### LINKED ISSUE #10310: Explore improved API Organization for fluid-framework package
One of the key issues with our API docs is that https://fluidframework.com/docs/apis/fluid-framework/ folds and re-exports four different packages, causing our generated API docs to lack organization and clarity. With this item we are looking for ways to organize fluid-framework api docs, to better communicate layering or grouping of APIs it exports. What we learn out of these explorations we can apply to other packages, but the goal for this particular ticket is to address `fluid-framework` package only.

### PR DESCRIPTION
**Pull request by ssimic2**
Closes https://github.com/microsoft/FluidFramework/issues/10310.

`fluid-framework` folds and re-exports four different packages, causing generated API docs to lack organization and clarity. Furthermore, fluid-framework does not give reader any deeper context about APIs it exports other than stating it "bundles a collection of Fluid Framework client packages for easy use". With this PR we are splitting `fluid-framework` docs based on exported packages. We are keeping it transparent what packages are folded within fluid-framework and we are letting the user drill into api docs for those packages, as they explore fluid-framework.

This seems like the most natural logical boundary/grouping we can use to improve `fluid-framework` api doc organization, without needing to create another, more synthetic, categorization schema. Each sub-package offers the reader further context they can use to reason about the APIs and functionality they offer. In the long term, if we decide to re-export further packages from fluid-framework, they will get their own api docs grouping with some minor config changes. On the other hand, if we decide to get rid of fluid-framework and export packages individually at some point in time, being transparent about the re-exported packages from the get-go, may help us present that direction change, as a smaller, somewhat natural progression. 

Screenshot 2022-05-22 182429

### CODE REVIEW

File: docs/_replacements/docs/apis/fluid-framework.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: "fluid-framework Package"
+kind: "Package"
+type: apis
+package: "fluid-framework"
+unscopedPackageName: "fluid-framework"
+area: "libraries"
+summary: "The fluid-framework package bundles a collection of Fluid Framework client packages for easy use when paired with a corresponding service client package (for example, @<!-- -->fluidframework/azure-client or @<!-- -->fluidframework/tinylicious-client)."
```
Comment (ssimic2) [inline docs/_replacements/docs/apis/fluid-framework.md:8]: @tylerbutler Let me know if you have better idea here. Our original fluid-framework.md is shadowed by this file now, so getting `.Site.Pages` data out of it, did not seem as feasible option.

File: docs/rollup-api-data.js [hunk 2]
```diff
@@ -42,12 +46,10 @@ const memberCombineInstructions = [
         ])
     },
     {
-        package: "fluid-framework",
+        package: "@fluidframework/container-definitions",
+        cleanOrgMembers: true,
```
Comment (tylerbutler) [inline docs/rollup-api-data.js:50]: nit: I read "org" as "organization." Consider "orig" instead.

File: docs/_replacements/docs/apis/fluid-framework.md (hunk 1 above)
Comment (tylerbutler) [reply docs/_replacements/docs/apis/fluid-framework.md:8]: Yeah, this is a fine solution I think.

Comment (tylerbutler) [reply docs/_replacements/docs/apis/fluid-framework.md:8]: However, I happened to be testing with a more recent version of hugo, and in those, the table of packages doesn't render. I think the mount resolution has changed in recent versions, but a fix that seems to work across versions is to rename this file to be `_replacements\docs\apis\fluid-framework\_index.md`.

**Review by tylerbutler — APPROVED** [review]
I love the overall simplicity of this change, but it is a reminder that the way the navigation works in the site template is RIDICULOUS and complex.

File: docs/_replacements/docs/apis/fluid-framework.md (hunk 1 above)
Comment (ssimic2) [reply docs/_replacements/docs/apis/fluid-framework.md:8]: Good find!

**Comment by ssimic2** [comment]
> I love the overall simplicity of this change, but it is a reminder that the way the navigation works in the site template is RIDICULOUS and complex.

I'm getting familiar with templating here but generally agree - not too much flexibility there as well.

**Pull request merged**